### DISCUSSION TITLE
Payment requests improvements for invoicing

### DISCUSSION THREAD

**Discussion by NicolasDorier** [btcpayserver/btcpayserver#2500] [General]
I want to continue improving payment requests.

I think the payment history is wrong, for example look at this:

![image](https://user-images.githubusercontent.com/3020646/116838402-33e10480-ac09-11eb-8c37-450680c01cef.png)

What you see is not the payment history, it is the history of BTCPay invoices.

From accountant point of view there is lot's of ambiguity:
* This page represents for him an invoice, yet, he can see several invoice id in a single payment request
* There is a line in "payment history" while no payment has been made.

I think that we should rather show the actual payment.

* How much money was actually sent?
* At which rate?
* At which address?
* At what time?

@dennisreimann @britttttk @dstrukt

**Comment by britttttk** [comment]
This is how it looks when payment is received @NicolasDorier 

![post-payment](https://user-images.githubusercontent.com/39231115/116839742-ae217280-ab90-11eb-8f07-e9d65bcd5019.png)

**Comment by Kukks** [comment]
Perhaps we can have a display toggle that only shows payments vs the invoices + payments.

Imo it can be important to show the invoices too to see if they tried to game the rate to pay the invoice, if they underpaid in parts of it, etc.

**Reply by britttttk** [reply]
yes I was thinking an expander too @Kukks

**Reply by NicolasDorier** [reply]
@Kukks yeah, but on the invoice print for the accountant, the invoices are just confusing.

**Comment by dstrukt** [comment]
Will take a look at this later today and think on some solutions, agreed that it needs improving, and there's plenty of room to do so.

I've also had a few simple ideas to improve the existing UI as it currently stands as well, so maybe we can work these in at the same time.

**Comment by pavlenex** [comment]
Had a chat with @NicolasDorier and ideally, the print invoice should remove all the confusing information and show only what may be relevant for the accountant.

It's been suggested that we remove the invoice IDs in the payment preview and show only what’s actually been paid.
